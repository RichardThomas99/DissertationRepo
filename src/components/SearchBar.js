import React, {Component} from 'react';
import'../App.css';

class SearchBar extends Component
{
  clicked()
  {
    var searchTerm = document.getElementById("textSearch").value;
    console.log("Search Term: " + searchTerm);
  }

render()
{
  return(
    <div class = "searchDiv">
      <form>
        <label>
          <input id = "textSearch"  type="text" name="product" placeholder ="Search a Product..." />
        </label>
          <button onClick = {(e) => {e.preventDefault(); this.clicked();}}> Submit </button>
        </form>
      </div>
    );
}
}

export default SearchBar;
