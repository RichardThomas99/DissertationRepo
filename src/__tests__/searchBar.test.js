import React from 'react';
import ReactDOM from 'react-dom';
import SearchBar from '../components/SearchBar';
import renderer from 'react-test-renderer'

it("SearchBar Render",() => {
  const searchBarComponent = renderer.create(
    <SearchBar/>
  ).toJSON();
  expect(searchBarComponent).toMatchSnapshot();
});
