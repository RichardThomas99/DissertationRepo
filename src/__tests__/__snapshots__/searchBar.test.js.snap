// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Render 1`] = `
<div
  class="searchDiv"
>
  <form>
    <label>
      <input
        id="textSearch"
        name="product"
        placeholder="Search a Product..."
        type="text"
      />
    </label>
    <button
      onClick={[Function]}
    >
       Submit Search 
    </button>
  </form>
</div>
`;
